pkg/standalone-utils/contracts/PriceImpactHelper.sol

```diff
@@ -0,0 +1,251 @@
+        return
+            abi.decode(
+                _callAndRevert(
+                    address(_router),
+                    abi.encodeWithSelector(
+                        _router.queryAddLiquidityUnbalanced.selector,
+                        pool,
+                        exactAmountsIn,
+                        sender,
+                        ""
+                    )
+                ),
+                (uint256)
+            );
+    }
+
+    function _queryRemoveLiquidityProportional(
+        address pool,
+        uint256 bptAmountOut,
+        address sender
+    ) internal returns (uint256[] memory) {
+        return
+            abi.decode(
+                _callAndRevert(
+                    address(_router),
+                    abi.encodeWithSelector(
+                        _router.queryRemoveLiquidityProportional.selector,
+                        pool,
+                        bptAmountOut,
+                        sender,
+                        ""
+                    )
+                ),
+                (uint256[])
+            );
+    }
+
+    function _querySwapSingleTokenExactIn(
+        address pool,
+        IERC20 tokenIn,
+        IERC20 tokenOut,
+        uint256 amountIn,
+        address sender
+    ) internal returns (uint256) {
+        return
+            abi.decode(
+                _callAndRevert(
+                    address(_router),
+                    abi.encodeWithSelector(
+                        _router.querySwapSingleTokenExactIn.selector,
+                        pool,
+                        tokenIn,
+                        tokenOut,
+                        amountIn,
+                        sender,
+                        ""
+                    )
+                ),
+                (uint256)
+            );
+    }
+
+    function _querySwapSingleTokenExactOut(
+        address pool,
+        IERC20 tokenIn,
+        IERC20 tokenOut,
+        uint256 amountOut,
+        address sender
+    ) internal returns (uint256) {
+        return
+            abi.decode(
+                _callAndRevert(
+                    address(_router),
+                    abi.encodeWithSelector(
+                        _router.querySwapSingleTokenExactOut.selector,
+                        pool,
+                        tokenIn,
+                        tokenOut,
+                        amountOut,
+                        sender,
+                        ""
+                    )
+                ),
+                (uint256)
+            );
+    }
+
+    /*******************************************************************************
+                                    Helpers
+    *******************************************************************************/
+
+    function _queryAddLiquidityUnbalancedForTokenDeltas(
+        address pool,
+        uint256 tokenIndex,
+        int256[] memory deltas,
+        address sender
+    ) internal returns (int256 deltaBPT) {
```

Comment:
I think I don't get the back and forth here. Why is this a signed value if we always change the result to be positive?